src/devices/src/virtio/net/device.rs

```diff
@@ -195,7 +229,7 @@ impl Net {
             activate_evt: EventFd::new(libc::EFD_NONBLOCK).map_err(Error::EventFd)?,
             config_space,
             mmds_ns: None,
-            guest_mac: guest_mac.copied(),
+            guest_mac: mac_addr,
 
             #[cfg(test)]
             mocks: Mocks::default(),
```

Comment:
I suggest you add a simple unit test where we just check that the result of generate_mac_address  gives back something different than KVM_OUI (which would mean that our rand code is working);

Reply:
Added a test in net.rs, that's where the unit test for adding a network device is.